server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1248,6 +1251,18 @@ private CanMatchResponse canMatch(ShardSearchRequest request, boolean checkRefre
                 if (canMatch || hasRefreshPending) {
                     FieldSortBuilder sortBuilder = FieldSortBuilder.getPrimaryFieldSortOrNull(request.source());
                     minMax = sortBuilder != null ? FieldSortBuilder.getMinMaxOrNull(context, sortBuilder) : null;
+
+                    if (RollupV2.isEnabled()) {
+                        IndexMetadata requestIndexMetadata = clusterService.state().getMetadata()
+                                .index(request.shardId().getIndexName());
```

Comment:
You can access the metadata of the index through `context.getIndexSettings().getIndexMetadata()`